### DISCUSSION TITLE
Getting same Vision Assistant style of object measurement image

### DISCUSSION THREAD

**Discussion by SnakePep** [ifm/ifm3d#440] [Q&A]
Is it possible to get same image, as Vision Assistant displays for object measurement (O3D354 Dimensioning of rectangular objects) - so rectangle put on the actual measurement image visualising what has been measured?

**Comment by lola-masson** [comment]
Hi @SnakePep,

I am not sure I get the full questions. Are you trying to visualize the measurement with the Vision Assistant, or with ifm3d? ifm3d does not support the O3D sensor, only the O3D camera, so you wouldn't be able to use the ifm3d library with an O3D354.

**Reply by graugans** [reply]
As @lola-masson mentioned the ifm3d library is build for cameras, nevertheless it is also possible to use the sensors with the ifm3d library. To retrieve the results one can use the [JSON_MODEL](https://github.com/ifm/ifm3d/blob/838292e016ffc29da20bf6896ffd8bc21a9dee3e/modules/framegrabber/include/ifm3d/fg/frame.h#L145) to retrieve the complete result data. Once you have extracted the information you are looking for you can create a similar visualization like the Vision Assistant does.

**Reply by SnakePep** [reply]
@lola-masson currently I'm just basically sending trigger and getting result via TCP -this works fine. Ultimate goal is to get image (same seen in VA) from the moment measurement is taken. That's why I started looking for a solution as ifm3d lib readme show it's compatible with O3D.

**Reply by SnakePep** [reply]
@graugans Thank you for clarification. By any chance is there any example that could help me to start with?

**Reply by vksgaikwad3** [reply]
@SnakePep Please find the example code snippet which will help retrieve the `JSON_MODEL` for the `O3D sensor`
Please make sure you have an active application configured using IVA before you run the below example.

```cpp
#include <iostream>
#include <memory>
#include <string.h>

#include <ifm3d/fg.h>
#include <ifm3d/device/device.h>


int main(int argc, const char **argv)
{
  // get access to the camera
  auto cam = ifm3d::Device::MakeShared();
  ifm3d::FrameGrabber::Ptr fg = std::make_shared<ifm3d::FrameGrabber>(cam);
  
  // set buffer_id to receive the image data
  fg->Start({ifm3d::buffer_id::JSON_MODEL});

  auto frame = fg->WaitForFrame();
  
  if (frame.wait_for(std::chrono::milliseconds(1000)) != std::future_status::ready)
    {
      std::cerr << "Timeout waiting for camera!" << std::endl;
      return -1;
    }
  auto json_model = frame.get()->GetBuffer(ifm3d::buffer_id::JSON_MODEL);
  
  std::string json = std::string(json_model.ptr<char>(0), strnlen_s(json_model.ptr<char>(0), json_model.size()));
  std::cout << "JSON MODEL : \n" << json << std::endl;
   
  return 0;
}
```
